Source/WebInspectorUI/UserInterface/Views/Layers3DContentView.js

```diff
@@ -284,11 +296,88 @@ WI.Layers3DContentView = class Layers3DContentView extends WI.ContentView
     _createLayerGroup(layer) {
         let layerGroup = new THREE.Group;
         layerGroup.userData.layer = layer;
-        layerGroup.add(this._createLayerMesh(layer.bounds), this._createLayerMesh(layer.compositedBounds, true));
+
+        let fillMesh = this._createLayerMesh(layer.bounds, false, null);
+        let outlineMesh = this._createLayerMesh(layer.compositedBounds, true, null);
+        layerGroup.add(fillMesh, outlineMesh);
+
+        if (this._showLayerContent) {
+            this._loadLayerTexture(layer, (texture) => {
+                if (!texture)
+                    return;
+
+                if (!layerGroup.parent) {
+                    texture.dispose();
+                    return;
+                }
+
+                layerGroup.remove(fillMesh);
+                if (fillMesh.geometry)
+                    fillMesh.geometry.dispose();
+                if (fillMesh.material) {
+                    if (fillMesh.material.map)
+                        fillMesh.material.map.dispose();
+                    fillMesh.material.dispose();
+                }
+
+                let texturedMesh = this._createLayerMesh(layer.bounds, false, texture);
+                layerGroup.add(texturedMesh);
+
+                layerGroup.remove(outlineMesh);
+                layerGroup.add(outlineMesh);
+            });
+        }
+
         return layerGroup;
     }
 
-    _createLayerMesh({x, y, width, height}, isOutline = false)
+    _loadLayerTexture(layer, callback) {
+        if (this._pendingTextureLoads.has(layer.layerId))
+            this._pendingTextureLoads.get(layer.layerId).cancelled = true;
+
+        let loadContext = { cancelled: false };
+        this._pendingTextureLoads.set(layer.layerId, loadContext);
+
+        WI.layerTreeManager.snapshotForLayer(layer, (dataURL) => {
```

Comment:
for complex pages this will result in a _lot_ of protocol traffic happening at once

i wonder if we should add a `requestContent` parameter to `LayerTree.layersForNode` that populates a new `content` property on the `LayerTree.Layer` object?

though this could mean that the response for `LayerTree.layersForNode` is gigantic due to having so much base64 data in it

id be curious to know which approach is faster and/or more efficient